core/platform/bulk_email/mailchimp_bulk_email_services.py

```diff
@@ -198,22 +214,44 @@ def add_or_update_user_status(
         'status': 'unsubscribed'
     }
 
+    tag_data = {
+        'tags': [{
+            'name': tag,
+            'status': 'active'
+        }]
+    }
+
+    # Additional fields for the Android tag.
+    if tag == 'Android':
+        new_user_mailchimp_data = {
+            'email_address': user_email,
+            'status': 'subscribed',
+            'tags': [tag],
+            'NAME': merge_fields['NAME']
+        }
+        subscribed_mailchimp_data = {
+            'email_address': user_email,
+            'status': 'subscribed',
+            'NAME': merge_fields['NAME']
+        }
+
     try:
-        member_details = client.lists.members.get(
+        _ = client.lists.members.get(
             feconf.MAILCHIMP_AUDIENCE_ID, subscriber_hash)
 
         # If member is already added to mailchimp list, we cannot permanently
         # delete a list member, since they cannot be programmatically added
         # back, so we change their status based on preference.
-        if (
-                can_receive_email_updates and
-                member_details['status'] != 'subscribed'):
+        if can_receive_email_updates:
+            # Here we use MyPy ignore because the following is a function in the
```

Comment:
Ditto. We should update the stubs I guess?

Reply:
Done.